awsiot/greengrasscoreipc/model.py

```diff
@@ -50,6 +50,15 @@ def __init__(self, *,
         self.memory = memory  # type: typing.Optional[int]
         self.cpus = cpus  # type: typing.Optional[float]
 
+    def set_memory(self, memory: int):
+        self.memory = memory
+        return self
```

Comment:
"builder" pattern isn't very common in python

users could already do:
```
limits = SystemResourceLimits(memory=1024, cpus=2)

# OR 
limits = SystemResourceLimits(memory=1024,
                              cpus=2)

# OR
limits = SystemResourceLimits()
limits.memory = 1024
limits.cpus = 2
```

I'm not sure these new functions help at all. These doesn't seem any more ergonomic to me:
```
limits = SystemResourcesLimits().set_memory(1024).set_cpus(2)

# OR 
limits = SystemResourcesLimits() \
    .set_memory(1024) \
    .set_cpus(2)
```

Reply:
it is more convenient when passing around an object, such as when one method fills in some fields using the constructor, and then another method needs to add more things. In that case, the builder is nicer to use given chained calls.